thunder/torch/__init__.py

```diff
@@ -2844,6 +2844,39 @@ def _normalize(a: TensorProxy, /, norm_dims, eps: Number) -> tuple[TensorLike, T
     return out, mean, rstd
 
 
+@torchsymbol(torch.nn.functional.normalize, is_method=True)
+def normalize(
+    a: TensorProxy, /, p: float = 2.0, dim: int | Sequence[int] = 1, eps: float = 1e-12, out: None | TensorProxy = None
+) -> TensorProxy:
+    utils.check(
+        dtypes.is_float_dtype(a.dtype) or dtypes.is_complex_dtype(a.dtype),
+        lambda: f"normalize: Expected a floating point or complext tensor as input. Got {a.dtype}",
+        TypeError,
+    )
+    utils.check(out is None, lambda: "normalize: out is not None which is currently unsupported", NotImplementedError)
+    computation_dtype, result_dtype = _reduction_dtypes(a, REDUCTION_OUTPUT_TYPE_KIND.COMPLEX_TO_FLOAT, a.dtype)
+    if p == 0.0:
+        denom = sum(a != 0.0, dim=dim, keepdim=True)
+    elif p == float("inf"):
+        denom = amax(abs(a), dim=dim, keepdim=True)
+    elif p == -float("inf"):
+        denom = amin(abs(a), dim=dim, keepdim=True)
+    else:
+        dim = utils.canonicalize_dim(a.ndim, dim)
+        a_ = clang.maybe_convert_to_dtype(a, computation_dtype)
+        is_p_even = p % 2.0 == 0
+        if not (is_p_even and dtypes.is_float_dtype(a.dtype)):
+            a_ = abs(a_)
```

Comment:
nit: probably should be `a <- abs(a) if (complex or not is_p_even)`?

Reply:
Yeah I can definitely do that! Is there a reason why we would prefer one or the other? Just curious :)

Reply:
Well, because for complex inputs we always have to call abs.

Reply:
Oh I think my question was, aren't the following two the same?
`if (complex or not is_p_even)` and `if not (is_p_even and float)`

Reply:
However, just incase I am missing anything, let me update the code :)

Reply:
If you grep over `is_float_type` you could see that it does not contain complex types.

Reply:
Oh, I see the point. Thanks for the detailed explanation. I naively thought `not float` means `complext` 😅 thank you so much for the help!